go/cmd/dolt/commands/filter-branch.go

```diff
@@ -104,9 +104,7 @@ func (cmd FilterBranchCmd) Exec(ctx context.Context, commandStr string, args []s
 		return HandleVErrAndExitCode(verr, usage)
 	}
 
-	if dEnv.IsLocked() {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(env.ErrActiveServerLock.New(dEnv.LockFile())), help)
-	}
+	// TODO: Assert that dEnv.DoltDB.AccessMode() != ReadOnly
```

Comment:
Yes, exactly, writes will fail if the implementation of the command attempts to do a write.

I did add a pre-check to `dolt gc`, because it can do a lot of work before it actually attempts the write, and that's a bad user experience.

I removed the TODOs in other places.